File: apps/api/src/app/exercise.entity.ts
import { Column, Entity, OneToMany, PrimaryGeneratedColumn } from 'typeorm';
import { ExerciseMuscleEntity } from './exerciseMuscle.entity';

@Entity({ name: 'exercise', schema: 'public' })
export class ExerciseEntity {
  @PrimaryGeneratedColumn('increment')
  id: number;

  @Column()
  name: string;

  @Column()
  description: string;

  @OneToMany(
    () => ExerciseMuscleEntity,
    (exerciseMuscle) => exerciseMuscle.id,
    { onDelete: 'CASCADE' }
  )
  exerciseMuscle: ExerciseMuscleEntity[];
}

File: my-workout-api/src/main/java/com/avernet/myworkoutapi/exercise/ExerciseEntity.java
package com.avernet.myworkoutapi.exercise;

import com.avernet.myworkoutapi.exercisemuscle.ExerciseMuscleEntity;
import com.avernet.myworkoutapi.history.HistoryEntity;
import com.avernet.myworkoutapi.userexercise.UserExerciseEntity;
import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "exercises")
@Getter
@Setter
public class ExerciseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long id;

    @Column()
    String name;

    @Column(columnDefinition = "TEXT")
    String description;

    @Column(name = "is_smart_workout")
    boolean isSmartWorkout;

    @OneToMany(mappedBy = "exercise", fetch = FetchType.LAZY, cascade = CascadeType.ALL, orphanRemoval = true)
    List<ExerciseMuscleEntity> exerciseMuscles = new ArrayList<>();

    @OneToMany(mappedBy = "exercise", fetch = FetchType.LAZY, cascade = CascadeType.ALL, orphanRemoval = true)
    List<HistoryEntity> histories = new ArrayList<>();

    @OneToMany(mappedBy = "exercise", fetch = FetchType.LAZY, cascade = CascadeType.ALL, orphanRemoval = true)
    List<UserExerciseEntity> userExercises = new ArrayList<>();
}
